logging/grafana-loki/docker-compose.yml

```diff
@@ -0,0 +1,114 @@
+version: "3.7"
+
+networks:
+  traefik-proxy:
+    external: true
+  loki:
+
+services:
+  loki:
+    image: grafana/loki:2.9.0
+    ports:
+      - "3100:3100"
+    command: -config.file=/etc/loki/loki-config.yaml
+    labels:
+      - traefik.enable=true
+      - traefik.http.services.loki.loadbalancer.server.port=3100
+      - traefik.http.routers.loki.rule=Host(`logging.subspace.network`)
+      - traefik.http.routers.loki.tls.certresolver=le
+      - traefik.http.routers.loki.entrypoints=websecure
+      - traefik.docker.network=traefik-proxy
+    volumes:
+      - /root/loki/conf/loki/etc/loki-config.yaml:/etc/loki/loki-config.yaml:ro
+    networks:
+      - loki
+      - traefik-proxy
+
+  promtail:
+    image: grafana/promtail:2.9.0
+    command: -config.file=/etc/promtail/config.yml
+    ports:
+      - 127.0.0.1:3200:3100
+    networks:
+      - loki
+
+  mimir:
+    image: grafana/mimir:2.9.0
+    volumes:
+      - /root/mimir/conf/etc/mimir.yml:/etc/mimir-config/mimir.yml:ro
+    entrypoint:
+      - /bin/mimir
+      - -config.file=/etc/mimir-config/mimir.yml
+    ports:
+      - "9009:9009"
+    networks:
+      - loki
+
+  agent:
+    profiles: [agent]
+    image: grafana/agent:latest
+    volumes:
+      - /root/agent/conf/etc/agent.yml:/etc/agent-config/agent.yml:ro
+    entrypoint:
+      - /bin/grafana-agent
+      - -server.http.address=0.0.0.0:12345
+      - -config.file=/etc/agent-config/agent.yml
+      - -metrics.wal-directory=/tmp/agent/wal
+      - -enable-features=integrations-next
+      - -config.expand-env
+      - -config.enable-read-api
+    environment:
+      HOSTNAME: agent
+      REMOTE_WRITE_HOST: mimir:9009
+      LOKI_HOST: loki:3100
+    ports:
+      - "12345:12345"
+    networks:
+      - loki
+    depends_on:
+      - mimir
+      - loki
+
+  grafana:
+    environment:
+      - GF_PATHS_PROVISIONING=/etc/grafana/provisioning
+      - GF_AUTH_ANONYMOUS_ENABLED=true
+      - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
+    entrypoint:
+      - sh
+      - -euc
+      - |
+        mkdir -p /etc/grafana/provisioning/datasources
+        cat <<EOF > /etc/grafana/provisioning/datasources/ds.yaml
+        apiVersion: 1
+        datasources:
+        - name: Loki
+          type: loki
+          access: proxy
+          orgId: 1
+          url: http://loki:3100
+          basicAuth: false
```

Comment:
How is the auth setup if the basic auth is disabled ?

Reply:
Grafana Auth dashboard over TLS. No need for basic auth since they are on the same machine, and not accessing loki externally.